test/Microsoft.Dnx.Testing.CommonUtils/TestUtils.cs

```diff
@@ -0,0 +1,99 @@
+            }
+            else if (string.Equals("coreclr", flavor, StringComparison.OrdinalIgnoreCase))
+            {
+                return FrameworkNameHelper.ParseFrameworkName("dnxcore50");
+            }
+
+            throw new InvalidOperationException($"Unknown runtime flavor '{flavor}'");
+        }
+
+        public static Solution GetSolution<TTest>(
+            DnxSdk sdk, 
+            string solutionName, 
+            [CallerMemberName]string testName = null, 
+            bool appendSolutionNameToTestFolder = false)
+        {
+            var rootPath = ProjectResolver.ResolveRootDirectory(Directory.GetCurrentDirectory());
+            var originalSolutionPath = Path.Combine(rootPath, "misc", solutionName);
+            var tempSolutionPath = GetTestFolder<TTest>(sdk, testName + (appendSolutionNameToTestFolder ? $".{solutionName}" : string.Empty));
+            CopyFolder(originalSolutionPath, tempSolutionPath);
+            return new Solution(tempSolutionPath);
+        }
+
+        public static string GetTempTestFolder<T>(DnxSdk sdk, [CallerMemberName]string testName = null)
+        {
+            return GetTestFolder<T>(sdk, $"{testName}.{Path.GetRandomFileName()}");
+        }
+
+        public static string GetTestFolder<T>(DnxSdk sdk, [CallerMemberName]string testName = null)
+        {
+            // This env var can be set by VS load profile
+            var basePath = Environment.GetEnvironmentVariable("DNX_LOCAL_TEMP_FOLDER_FOR_TESTING");
+            if (string.IsNullOrEmpty(basePath))
+            {
+                var rootPath = ProjectResolver.ResolveRootDirectory(Directory.GetCurrentDirectory());
+                basePath = Path.Combine(rootPath, "TestOutput");
+            }
+
+            var tempFolderPath = Path.Combine(basePath, sdk.FullName, $"{typeof(T).Name}.{testName}");
+            Directory.CreateDirectory(tempFolderPath);
+            return tempFolderPath;
+        }
+
+        public static void CopyFolder(string sourceFolder, string targetFolder)
+        {
+            if (!Directory.Exists(targetFolder))
+            {
+                Directory.CreateDirectory(targetFolder);
+            }
+
+            foreach (var filePath in Directory.EnumerateFiles(sourceFolder))
+            {
+                var fileName = Path.GetFileName(filePath);
+                File.Copy(filePath, Path.Combine(targetFolder, fileName));
+            }
+
+            foreach (var folderPath in Directory.EnumerateDirectories(sourceFolder))
+            {
+                var folderName = new DirectoryInfo(folderPath).Name;
+                CopyFolder(folderPath, Path.Combine(targetFolder, folderName));
+            }
+        }
+
+        public static string CreateLocalFeed<TTest>(DnxSdk sdk, Solution solution, [CallerMemberName]string testName = null)
+        {
+            var feed = GetTestFolder<TTest>(sdk, $"{testName}.{Path.GetRandomFileName()}");
```

Comment:
These temporary folders are created under the dnx root so they should not get into trouble with var - private/var on Mac